src/BloomExe/Publish/Epub/PublishEpubApi.cs

```diff
@@ -161,6 +152,35 @@ public void RegisterWithApiHandler(BloomApiHandler apiHandler)
 		}
 
 		private void HandleEpubSave(ApiRequest request)
+		{
+			if (ControlForInvoke != null && ControlForInvoke.InvokeRequired)
+			{
+				ControlForInvoke.Invoke((Action)(() => getSaveAsPath(request)));
+			}
+			else
+			{
+				getSaveAsPath(request);
+			}
+						
+			// If we ARE in the middle of staging the epub...quite possible since this
+			// handler is registered with permission to execute in parallel with other
+			// API handlers, the user just has to click Save before the preview is finished...
+			// then we need not do any more here. A call to SaveAsEpub at the end of the
+			// preview generation process will pick up the pending request in _pendingSaveAsPath
+			// and complete the Save. 
+			// 
+			// Otherwise, RefreshPreview will call SaveAsEpub which will pick up the 
+			// _pendingSaveAsPath and save
+
+			if (!_stagingEpub && _pendingSaveAsPath != null) // this should only happen if an output path was chosen
```

Comment:
Should this block take the `_epubMakerLock`? I think it is not necessary, as worst case we run `RefreshPreview` unnecesarily. But I may not have thought of all the race conditions. `RefreshPreview` is a pretty long block to put in a lock